plugin/al/usdmayautils/AL/usdmaya/utils/DgNodeHelper.cpp

```diff
@@ -2187,7 +2187,8 @@ MStatus DgNodeHelper::getVec3Array(MObject node, MObject attribute, GfHalf* cons
     v[0] = elem.child(0).asFloat();
     v[1] = elem.child(1).asFloat();
     v[2] = elem.child(2).asFloat();
-    AL::usd::utils::float2half_4f(v, h);
+    v[3] = 0;
```

Comment:
This is a de-optimisation. I honestly could not care less about -strict compiler warnings on this line of code. The code change is clearly worse. Remove this assignment.

Reply:
@robthebloke Use-before-set errors are tricky to find and debug, so I wouldn't recommend ignoring compiler warnings. Looking at the performance side, it as well tells you something. You are making a copy of 4 elements with the cast where the last one is not initialized. This means you don't even care about the last element. Given that SIMD is disabled (#122), the currently compiled code could be much more efficient if you would have only operated on 3 channels. This would as well avoid the compiler warning.

To close on it, we can't merge changes that don't compile in strict mode.